imhex: 1.19.3 -> 1.24.3

panicgh opened the pull request:
###### Description of changes

Package update for various bug fixes and improvements: https://github.com/WerWolv/ImHex/releases

I tried to build it with `USE_SYSTEM_CAPSTONE=ON`, but compilation failed. So I kept it switched `OFF`.

Opening a file and similar operations shows an error complaining about missing `xdg-desktop-portal` executables in the path, but drag & drop a file to the imhex GUI works. That is the [same  limitation](https://github.com/NixOS/nixpkgs/pull/184488#issuecomment-1207767466) as with the previous version. Maybe it is an issue on my system and I would have to set [xdg.portal.enable](https://github.com/NixOS/nixpkgs/blob/nixos-22.05/nixos/modules/config/xdg/portal.nix) in my NixOS config, but I am not sure.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).


Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>imhex</li>
  </ul>
</details>

panicgh commented on pkgs/applications/editors/imhex/default.nix:
```diff
@@ -77,6 +77,7 @@ gcc12Stdenv.mkDerivation rec {
   postInstall = ''
     ln -s $out/share/imhex/plugins $out/bin/
```
The built-in plugin is found even when the symlink is not there. But I'm not sure if the files from `patterns_src` are correctly loaded (maybe they should go somewhere in `$out/lib/imhex` instead of `$out/share/imhex`).

```suggestion
```

SuperSandro2000 replied:
Also they should not be under $out/bin

dev-null-undefined committed:
imhex: 1.19.3 -> 1.26.2

Co-authored-by: x10an14 <x10an14@gmail.com>

dev-null-undefined commented:
@panicgh can you bump the version to 1.26.2
I have created a commit trying to do that, but I don't want to steal the PR from you https://github.com/dev-null-undefined/nixpkgs/commit/4d10293e5b3ee7aa45b2461bc6329f958a2002fe
Had to move some of the steps to `postBuild` phase otherwise the CMake install phase would try to create them in install phase resulting into fail with error `permission denied` on trying to create those directories.
Many thanks to @x10an14 for debugging that with me and helping me out.

panicgh commented:
I rebased the branch on master and used your proposed commit as base. Then I noticed that the symlink `$out/bin/plugins` is broken because `$out/share/imhex/plugins` did not exist. I changed it to `$out/lib/imhex/plugins`, but then imhex showed an error at startup that the same built-in plugin was loaded twice. So finally, I removed the `$out/bin/plugins` symlink altogether as it doesn't seem to be necessary (it is probably an *additional* location where plugins can be loaded from).

The `postBuild` step works as you suggested, but so does it -- for me at least on x86_64-linux -- with `postInstall`, too. I do not get an error. Please try to build this and let me know if you still get the permission error. If so, I'll change it back to `postBuild`.

dev-null-undefined commented:
Result of `nixpkgs-review pr 197762` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>imhex</li>
  </ul>
</details>

dev-null-undefined commented:
Tested it and it build without a problem thanks for the version bump tho @panicgh

Garmelon commented:
Once I uninstalled the older imhex that still put its builtin plugins in `/run/current-system/sw/share/imhex/plugins/`, this fixed #197109.
